pkg/apis/componentconfig/v1alpha1/types.go

```diff
@@ -292,25 +294,25 @@ type CloudControllerManagerConfiguration struct {
 	metav1.TypeMeta `json:",inline"`
 
 	// CloudProviderConfiguration holds configuration for CloudProvider related features.
-	CloudProvider CloudProviderConfiguration
+	CloudProvider CloudProviderConfiguration `json:"cloudProvider"`
 	// DebuggingConfiguration holds configuration for Debugging related features.
-	Debugging DebuggingConfiguration
+	Debugging DebuggingConfiguration `json:"debugging"`
 	// GenericComponentConfiguration holds configuration for GenericComponent
 	// related features both in cloud controller manager and kube-controller manager.
-	GenericComponent GenericComponentConfiguration
+	GenericComponent GenericComponentConfiguration `json:"genericComponent"`
 	// KubeCloudSharedConfiguration holds configuration for shared related features
 	// both in cloud controller manager and kube-controller manager.
-	KubeCloudShared KubeCloudSharedConfiguration
+	KubeCloudShared KubeCloudSharedConfiguration `json:"kubeCloudShared"`
 	// ServiceControllerConfiguration holds configuration for ServiceController
 	// related features.
-	ServiceController ServiceControllerConfiguration
+	ServiceController ServiceControllerConfiguration `json:"serviceController"`
 	// NodeStatusUpdateFrequency is the frequency at which the controller updates nodes' status
-	NodeStatusUpdateFrequency metav1.Duration
+	NodeStatusUpdateFrequency metav1.Duration `json:"nodeStatusUpdateFrequency"`
 }
 
 type CloudProviderConfiguration struct {
 	// Name is the provider for cloud services.
-	Name string `json:"cloudProvider"`
```

Comment:
same question, why change external serialization instead of changing the field name?

Reply:
I plan to fix field naming and structure later in https://github.com/kubernetes/kubernetes/pull/66993, but now just make stuff consistent to make it pass validation

Reply:
is the assumption that none of these serialized fields are currently used in supported config files? changing serialization tags is not something we should generally do. I'd rather see this validation errors fixed by changing the go field name if possible.